animation_nodes/nodes/text/decompose_text.py

```diff
@@ -0,0 +1,107 @@
+    bl_idname = "an_DecomposeTextNode"
+    bl_label = "Decompose Text"
+    errorHandlingType = "EXCEPTION"
+
+    includeWhiteSpaces : BoolProperty(name = "Include White Spaces", default = True,
+        update = propertyChanged)
+
+    def create(self):
+        self.newInput("Text", "Text", "text")
+        self.newInput("Font", "Font", "font")
+        self.newInput("Float", "Character Spacing", "charSpacing", value = 1, hide = True)
+        self.newInput("Float", "Word Spacing", "wordSpacing", value = 1, hide = True)
+        self.newInput("Float", "Line Spacing", "lineSpacing", value = 1, hide = True)
+        self.newInput("Text", "Alignment", "align", value = "LEFT", hide = True)
+
+        self.newOutput("Matrix List", "Transforms", "transforms")
+        self.newOutput("Text List", "Characters", "characters")
+        self.newOutput("Integer", "Length", "length")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "includeWhiteSpaces")
+
+    def execute(self, text, font, charSpacing, wordSpacing, lineSpacing, align):
+        self.validateAlignment(align)
+        fontID = self.getFontID(font)
+        fontRatio = self.getFontRatio(font, fontID)
+
+        return self.getCharTransforms(text, fontID, fontRatio, charSpacing, wordSpacing, lineSpacing, align)
+
+    def getCharTransforms(self, text, fontID, fontRatio, charSpacing, wordSpacing, lineSpacing, align):
+        allChars = []
+        allTransforms = Matrix4x4List()
+        for i, line in enumerate(text.splitlines()):
+            chars = []
+            transforms = []
+            comulativeWidth = 0
+            for char in line:
+                if self.includeWhiteSpaces or char not in whitespace:
+                    transforms.append(Matrix.Translation((comulativeWidth, -i * lineSpacing, 0)))
+                    chars.append(char)
+
+                comulativeWidth += self.getCharWidth(fontID, char, fontRatio, charSpacing, wordSpacing)
+
+            transforms = Matrix4x4List.fromValues(transforms)
+
+            if align == "CENTER":
+                offset = (-comulativeWidth + charSpacing / 2 - 0.5) / 2
+                transforms.transform(Matrix.Translation((offset, 0, 0)))
+            elif align == "RIGHT":
+                offset = -comulativeWidth + charSpacing / 2 - 0.5
+                transforms.transform(Matrix.Translation((offset, 0, 0)))
+
+            allChars += chars
+            allTransforms.extend(transforms)
+
+        return allTransforms, allChars, len(allChars)
+
+    def getCharWidth(self, fontID, char, fontRatio, charSpacing, wordSpacing):
+        width = blf.dimensions(fontID, char)[0] * fontRatio
+        if char == " ":
+            width *= wordSpacing
+        return width + charSpacing / 2 - 0.5
+
+    @lru_cache()
+    def getFontRatio(self, font, fontID):
+        data = bpy.data.curves.new("an_helper_font_curve", type = "FONT")
+        textObject = bpy.data.objects.new("an_helper_text_object", data)
```

Comment:
The `textObject` has to be removed as well.

Reply:
`BlendDataCurves.remove` removes the objects by default.

Reply:
I see. I think either you should remove the `textObject` explicitely before removing the curve, or add a comment.

Reply:
Alright.